### PR TITLE
[Merged by Bors] - feat: an abstract continuous functional calculus for non-unital algebras

### PR DESCRIPTION
**Pull request by j-loreaux**
---

- [x] depends on: #11219
- [x] depends on: #11220

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousFunction/NonUnitalFunctionalCalculus.lean [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+  rw [cfcₙ_comp' a f (r * ·), cfcₙ_const_mul_id a r]
+
+lemma cfcₙ_comp_star (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (star '' (σₙ R a)) := by cfc_cont_tac)
+    (hf0 : f 0 = 0 := by cfc_zero_tac) :
+    cfcₙ a (f <| star ·) = cfcₙ (star a) f := by
+  rw [cfcₙ_comp' a f star, cfcₙ_star_id a]
+
+lemma eq_zero_of_quasispectrum_eq_zero (h_spec : σₙ R a = {0}) (ha : p a := by cfc_tac) :
+    a = 0 := by
```
Comment (j-loreaux) [inline Mathlib/Topology/ContinuousFunction/NonUnitalFunctionalCalculus.lean:376]: We can just use subset here
```suggestion
lemma eq_zero_of_quasispectrum_eq_zero (h_spec : σₙ R a ⊆ {0}) (ha : p a := by cfc_tac) :
    a = 0 := by
```

File: Mathlib/Topology/ContinuousFunction/NonUnitalFunctionalCalculus.lean [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+the Stone-Weierstrass theorem. -/
+class UniqueNonUnitalContinuousFunctionalCalculus (R A : Type*) [Semifield R] [StarRing R]
+    [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R] [NonUnitalRing A] [StarRing A]
+    [TopologicalSpace A] [Module R A] [IsScalarTower R A A] [SMulCommClass R A A] : Prop where
+  eq_of_continuous_of_map_id (s : Set R) [CompactSpace s] [Zero s] (h0 : (0 : s) = (0 : R))
+    (φ ψ : C(s, R)₀ →⋆ₙₐ[R] A) (hφ : Continuous φ) (hψ : Continuous ψ)
+    (h : φ (⟨.restrict s <| .id R, h0⟩) = ψ (⟨.restrict s <| .id R, h0⟩)) :
+    φ = ψ
+  compactSpace_qausispectrum (a : A) : CompactSpace (σₙ R a)
+
```
Comment (dupuisf) [inline Mathlib/Topology/ContinuousFunction/NonUnitalFunctionalCalculus.lean:92]:
```suggestion
  compactSpace_quasispectrum (a : A) : CompactSpace (σₙ R a)

```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

**Pull request closed without merging**